### ISSUE TITLE
UnicodeDecodeError when parsing user agent

### ISSUE THREAD

**Issue by ojotoxy** [jotes/django-cookies-samesite#38]
* django-cookies-samesite version: 0.6.6
* Django version: 1.11
* Python version: 2.7
* Operating System: Linux

### Description

A user hit this error, and their user-agent is "

Mozilla/5.0 (Linux; Android 7.1.2; Moto E� POWER) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.101 Mobile Safari/537.36
"

(I'm not sure if that user agent came through properly in the error logs, but the cause is likely just a unicode char in the user agent.)

### What I Did

```
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb3 in position 6: ordinal not in range(128)
  File "django/core/handlers/base.py", line 131, in get_response
    response = middleware_method(request, response)
  File "django_cookies_samesite/middleware.py", line 78, in process_response
    user_agent_checker = UserAgentChecker(http_user_agent)
  File "django_cookies_samesite/user_agent_checker.py", line 22, in __init__
    user_agent_string if user_agent_string else ""
  File "ua_parser/user_agent_parser.py", line 239, in Parse
    "device": ParseDevice(user_agent_string, **jsParseBits),
  File "ua_parser/user_agent_parser.py", line 323, in ParseDevice
    device, brand, model = deviceParser.Parse(user_agent_string)
  File "ua_parser/user_agent_parser.py", line 210, in Parse
    model = MultiReplace(self.model_replacement, match)
  File "ua_parser/user_agent_parser.py", line 157, in MultiReplace
    _string = re.sub(r"\$(\d)", _repl, string)
  File "re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)

```

**Comment by jotes** [comment]
Thanks for the report @ojotoxy, I'll take a look into it!

**Comment by abdul-rafiu** [comment]
@jotes i see some special characters in the user agent string that is causing the issue. I ll try fixing it in the weekday if I get time otherwise I ll fix this over the weekend.

**Comment by jotes** [comment]
@abdul-rafiu I'm already working on that. I'll try to post the fix later (Right now, it's 9 AM for me, let's say that I'll start working again on it around 6 PM).

I think the best solution is stripping non-ASCII characters. As you said, HTTP Headers shouldn't have them.

**Comment by jotes** [comment]
@ojotoxy Hey... I'm sorry about the spam, but could you look at the latest release (0.7.0) and tell me if it solved the issue for you?
I had some problems with the upload to PYPI.

**Comment by ojotoxy** [comment]
@jotes  I think version 0.6.6 has disappeared from pypi, you might want to put it back in case anyone was depending on it

**Comment by ojotoxy** [comment]
I'm having a hard time verifying the fix as I cannot reproduce the issue yet on the old version with curl, so I can't say much yet. The error was only found in the error logs.

0.7.0 doesn't seem to break anything else though!

**Comment by jotes** [comment]
@schinckel I'm really sorry about that, the next release is `0.7.0` and it has the fix mentioned for this issue.
`0.6.6` was released pre-maturely, and without any further experience how to handle such cases, I took a quick (and bad) decision to just remove the package from PYPI.
As I learned later, there are better ways to handle such scenarios (yanking etc.) and I could cause some problems for people.
